src/helpers/diceRoller/diceRoller.test.ts

```diff
@@ -0,0 +1,40 @@
+import { diceRoller, diceStringParser } from './diceRoller';
+const isTrue = (input: unknown) => !!input;
+describe('diceRoller', () => {
+  // it("(is/can/has/does prefix optional) simple and short human-readable description of what's expected", () => {
+  // expect contains your function, with some params, e.g. expect(isTrue(false))
+  // jest provides some helpers that then can be used to inspect the result, e.g. .toBe(), 
+  // });
+  
+  it("doesn't break reality", () =>{
+    expect(isTrue(false)).toBe(false);
+    expect(isTrue(true)).toBe(true);
+  })
+
+  it('converts dice strings and then rolls the dice', () => {
+    expect(diceRoller('1d6')).toBeGreaterThanOrEqual(1);
+    expect(diceRoller('1d6')).toBeLessThanOrEqual(6);
+    expect(diceRoller('100d100')).toBeGreaterThanOrEqual(1);
+    expect(diceRoller('100d100')).toBeLessThanOrEqual(Math.pow(100, 100));
+  });
+});
+
+describe('diceStringParser', () => {
+  it('reads die strings and returns dieCount and dieType', () => {
+    expect(diceStringParser('1d6')).toEqual({ dieCount: 1, dieType: 6 });
+  });
+  // also can do any arbirtrary javascript code, one I tend to use often is forEach of a set of values
+  ['-1', '1', '100', '1000', '10000', '1e10', 'Infinity'].forEach((num) => {

```

Comment:
feels like half of these are not needed, probably best to only bother with the actual numbers, instead of exponentials and Infinity, since you lockdown the inputs anyhow